src/main/java/Duke.java

```diff
@@ -1,10 +1,239 @@
+                deleteTask(inputDetails);
+            } else  {
+                if (!isValidCommand(userCommand)) {
+                    continue;
+                }
+                //process Todo, event or deadline
+                processCommandWithException(userCommand, inputDetails);
+            }
+            printLine();
+        }
     }
+
+    private static void processCommandWithException(String userCommand, String inputDetails) {
+        try {
+            processUserRequest(userCommand, inputDetails);
+        } catch (TodoException e) {
+            printLine();
+            System.out.println(e.sendErrorMessage());
+        }
+    }
+
+    private static boolean isValidCommand(String userCommand) {
+        boolean isValid = true;
+        try {
+            isValidInput(userCommand);
+        } catch (InvalidCommandException e) {
+            printLine();
+            System.out.println(e.sendErrorMessage());
+            printLine();
+            isValid = false;
+        }
+        return isValid;
+    }
+
+    private static void isValidInput(String userCommand) throws InvalidCommandException {
+        var isValid = userCommand.equalsIgnoreCase("todo") | userCommand.equalsIgnoreCase("deadline") | userCommand.equalsIgnoreCase("event");
+        if (!isValid) {
+            throw new InvalidCommandException();
+        }
+    }
+
+    private static void sendWelcomeMessage() {
+        System.out.println("Hello from\n" + LOGO);
+        printLine();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+    }
+
+    private static void createFileIfThereIsNone() {
+        File fForCheck = new File(FILE_PATH_TO_SAVE_TASKS);
+        if (!fForCheck.exists()) {
+            createNewFile(fForCheck);
+        }
+    }
+
+    private static void emptyFileAfterInitializing() {
+        FileWriter fw = null;
+        fw = createFileWriterObject(null);
+        writeEmptyStringToFile(fw);
+    }
+
+
+    private static void createNewFile(File fForCheck) {
+        try {
+            fForCheck.createNewFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static FileWriter createFileWriterObject(FileWriter fw) {
+        try {
+            fw = new FileWriter(FILE_PATH_TO_SAVE_TASKS);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return fw;
+    }
+
+    private static void writeEmptyStringToFile(FileWriter fw) {
+        try {
+            fw.write("");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void processUserRequest(String userCommand, String inputDetails) throws TodoException {
+        if (inputDetails.equalsIgnoreCase("filler") & userCommand.equalsIgnoreCase("todo")) {
+            throw new TodoException();
+        }
+        printLine();
+        printLine();
```

Comment:
It seems you have repeated lines of code here. Consider reviewing what is [redundant](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#delete-dead-code).